### PR TITLE
Fixed: Missing substring should raise ValueError not KeyError + try-except code cleanup

### PR DESCRIPTION
**Pull request by kchawla-pi**
Fix for issue #2690

### CODE REVIEW

**Comment by djarecka** [comment]
@kchawla-pi - thank you for the PR!

File: nipype/utils/spm_docs.py [hunk 1]
```diff
@@ -59,5 +59,5 @@ def _strip_header(doc):
         except ValueError:
             index = len(doc)
         return doc[:index]
-    except KeyError as e:
+    except ValueError as e:
```
Comment (effigies) [inline nipype/utils/spm_docs.py]: I think this whole block might be better rewritten as

```Python
    try:
        index = doc.index(hdr)
    except ValueError as e:
        raise_from(IOError('This docstring was not generated by Nipype!\n'), e)
    index += len(hdr)
    index += 1
    doc = doc[index:]
    try:
        index = doc.index(cruft)
    except ValueError:
        index = len(doc)
    return doc[:index]
```

This helps make clear where the exception is expected, and avoids catching unexpected exceptions.

**Comment by effigies** [comment]
Hi @kchawla-pi. Do you have a few minutes to finish this one off? I think we're almost ready to go.

We'll be releasing 1.1.3 next week, and this would be a great addition!

**Comment by kchawla-pi** [comment]
Ok.

**Pull request merged**